### PR TITLE
Refactor ws-daemon and registry-facade postStart hook

### PR DESCRIPTION
**Pull request by aledbf**
## Description

Refactor `postStart` and `preStop` hook to use a custom binary instead of external dependencies (`wait4x` and `kubectl`)

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/11514

## How to test
- Workspaces should open and work as expected

## Release Notes
```release-note
NONE
```

## Werft options:
- [x] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: dev/ready-probe-labeler/main.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	Shutdown   bool
+	Kubeconfig string
+	Label      string
+	ProbeURL   string
+	Timeout    time.Duration
+}
+
+func main() {
+	pflag.BoolVarP(&opts.JSONLog, "json-log", "j", true, "produce JSON log output on verbose level")
+	pflag.BoolVarP(&opts.Shutdown, "shutdown", "s", false, "Removes the node label previous to a pod stop")
```
Comment (sagor999) [inline dev/ready-probe-labeler/main.go:45]:
```suggestion
	pflag.BoolVarP(&opts.Shutdown, "shutdown", "s", false, "Removes the node label prior to a pod stop")
```

File: dev/ready-probe-labeler/main.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	if opts.Shutdown {
+		err = updateLabel(opts.Label, false, nodeName, client)
+		if err != nil {
+			log.Fatalf("Unexpected error removing node label: %v", err)
+		}
+
+		return
+	}
+
+	time.Sleep(1 * time.Second)
```
Comment (sagor999) [inline dev/ready-probe-labeler/main.go:80]: what is the reason for 1s sleep here?

File: dev/ready-probe-labeler/main.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	start := time.Now()
+
+	err = waitForURLToBeReachable(opts.ProbeURL, opts.Timeout)
+	if err != nil {
+		log.Fatalf("Unexpected error waiting for URL: %v", err)
```
Comment (sagor999) [inline dev/ready-probe-labeler/main.go:85]:
```suggestion
		log.Fatalf("Unexpected error waiting for probe URL: %v", err)
```

File: dev/ready-probe-labeler/main.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+	}
+
+	// do not set the label immediately
+	time.Sleep(10 * time.Second)
+
+	err = updateLabel(opts.Label, true, nodeName, client)
+	if err != nil {
+		log.Fatalf("Unexpected error waiting for URL: %v", err)
```
Comment (sagor999) [inline dev/ready-probe-labeler/main.go:93]:
```suggestion
		log.Fatalf("Unexpected error while trying to add the label: %v", err)
```

**Review by sagor999 — APPROVED** [review]
LGTM
/hold
for some minor nits.

File: dev/ready-probe-labeler/main.go (hunk 1 above)
Comment (aledbf) [reply dev/ready-probe-labeler/main.go]: fixed

File: dev/ready-probe-labeler/main.go (hunk 3 above)
Comment (aledbf) [reply dev/ready-probe-labeler/main.go]: fixed

File: dev/ready-probe-labeler/main.go (hunk 4 above)
Comment (aledbf) [reply dev/ready-probe-labeler/main.go]: fixed

File: dev/ready-probe-labeler/main.go (hunk 2 above)
Comment (aledbf) [reply dev/ready-probe-labeler/main.go:80]: No particular reason besides not starting right away

**Pull request merged**